app/scripts/controllers/pod.js

```diff
@@ -86,7 +86,115 @@ angular.module('openshiftConsole')
         });
       }
     };
+       
+    $scope.onChange = function(arg) {     
+      // Iter. over all containers, setting new to active/connected & old's isActive to false
+      _.forEach($scope.containerTerminals, function(ctr){
+        if (ctr === arg) {
+          ctr.isActive = true;
+          
+          // If not previously connected to (Prevents duplicates)
+          if (!ctr.isConnected) {
+            ctr.isConnected = true;
+            // Add this to the list of connected terminals
+            $scope.connectedTerminals.push(ctr);
+          }
+        } else {
+          ctr.isActive = false;
+        }
+      }); 
+    }
+    
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+    
+    var updateSelectTerminals = function() {
+      var activeList = _.filter($scope.containerTerminals, function(term) {
+          return term.isActive;
+      });
+      
+      $scope.connectedTerminals = _.filter($scope.containerTerminals, function(term) {
+        return term.isConnected;
+      });
+      
+      // Only make changes if there are no current actives
+      if (activeList.length <= 0) {
+        // If none connected, pick first of all containerTerminals (Could select a dead terminal)
+        if ($scope.connectedTerminals.length <= 0) {
+          _.head($scope.containerTerminals).isActive = true;
+          _.head($scope.containerTerminals).isConnected = true;
+          
+          console.log($scope.containerTerminals);
+          
+          updateSelectTerminals();
+        } else {
+          $scope.selected = _.head($scope.connectedTerminals).name;
+          $scope.selected.isActive = true;
+          $scope.selected.isConnected = true;
+        }
+      } else {
+        $scope.selected = (_.head(activeList));
+      }
+    }
+    
+    var populateFrontEndTerminalData = function(containers) {
+      console.log(containers);
+      
+      var availableTerminals = _.map(containers, function(container) {
+        return {
+          name: container.name,
+          isActive: false,
+          isConnected: false,
+          died: false
+        };
+      });
+      
+      availableTerminals.push({
+        name: 'testDead',
+        isActive: false,
+        isConnected: false,
+        died: true
+      });
+            
+      // INTERSECTION
+      _.forEach(availableTerminals, function(newTerm, outerIndex) {
+        
+        var index = _.findIndex($scope.containerTerminals, function(currentTerm) {
```

Comment:
If you went with `_.find()`, you can then also update the term you found like this:
```javascript
if(found) {
  angular.extend(availableTerm, found);   // will merge the objects, 2nd objects properties win.
}
```
This eliminates need to manually map the properties.